### ISSUE TITLE
output exe become unuseable

### ISSUE THREAD

**Issue by iNViTiON** [yck1509/ConfuserEx#278]
Some of my function not working correctly after protect with maximum preset. It happen every times I try to protect and run. I have to select to remove "ctrl flow" to allow my app work normally. Don't sure what information I have to tell you more and not sure where I have to check. If you need more info please ask me.

**Comment by MurariAlex** [comment]
Which version of ConfuserEX are you using?
Did you get it from the [Release Section](https://github.com/yck1509/ConfuserEx/releases), [CI Server](https://ci.appveyor.com/project/yck1509/confuserex/build/138/artifacts), or compiled yourself?

Try getting the version of the CI Server, it has several bugfixes, control flow related too.

**Comment by iNViTiON** [comment]
Yes. It was compiled bin 0.5.0 from here -> https://github.com/yck1509/ConfuserEx/releases

And..I don't know how to compile it myself. I try to download source from master branch before but not success (with build.cmd and VS). Then, I may have to waiting for next compiled version. Is this have a plan for new release or build instruction?

Thank you very much for this fast respond. At least I'm relief that it not my bad config.

**Comment by MurariAlex** [comment]
You don't need to compile it yourself. Download the latest binaries [here](https://ci.appveyor.com/project/yck1509/confuserex/build/138/artifacts).

**Comment by iNViTiON** [comment]
Oh. I looking on that page without notice that it have a binaries. Thank you.
Can I ask for one more? Why you prefer build 138 instead of build 140?

**Comment by MurariAlex** [comment]
The build number 139/140 comes from an [pull request](https://github.com/yck1509/ConfuserEx/pull/276) not evaluated by @yck1509 yet.
The build 138 is only a few hours old and is the last from @yck1509. :)

**Comment by iNViTiON** [comment]
Thank you. I think next time I will can select it by myself. :)

Even with build 138. My app become useable more than the one from v0.5.0 but some function still not working correctly. Too bad that I may release it without ctrl flow.

To be honest. I don't really understand "ctrl flow". How much it important compare to the other? At first I give the priority to Anti-debug and IL Dasm. (because it the only thing I know)

**Comment by MurariAlex** [comment]
As far as I am concerned, control flow is crucial to prevent that someone export your source to a VS project, using [DotPeek](https://www.jetbrains.com/decompiler/), for example, plus it makes real hard to read through the code.
Maybe you send to Ki (confuser.net@gmail.com) an example that demonstrates the issue and the crproj of ConfuserEx so he can analyze it.

**Comment by carlolars** [comment]
I have the same problem using 0.5.0.
We have rather large project written in VB.Net with dlls that we merge into a single exe using ILMerge that is then obfuscated using the CLI. I have identified it to be the ctrl flow that breaks the application and I have identified where in the code it breaks (may be more places, but this is in the startup of the application so I have not gotten any further).

When rebuilding (without modifying the source) I can get an application that crashes in one of three ways, and I can even sometimes get an application that works (at least gets past this first part of the code, there might be more errors in other parts that I have not discovered).

After some trial and error I have modified my code by wrapping one line of code with a try-catch and it always works (but no exception), at least the initial error is removed.

I have not been able to reduce the code that causes the problem, the application is quite complex with threads and serial-port communication and stuff that I have not even tried to understand.

**Comment by yck1509** [comment]
Hi,
If you're still having this problem, contact me at confuser.net@gmail.com so I could look into it.
